server/src/main.rs

```diff
@@ -0,0 +1,254 @@
+        Ok(serde_json::from_slice(body_bytes)?)
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    let app = AppState::new();
+
+    setup_app_dir().expect("App's directory setup failed");
+    let http = tokio::spawn(run_http(app.clone()));
+    let ws = tokio::spawn(run_ws(app.clone()));
+    let heartbeat = tokio::spawn(run_heartbeat_service(app.clone()));
+
+    heartbeat.await.expect("Heartbeat service died!");
+    ws.await.expect("WS server died!");
+    http.await.expect("HTTP server died!");
+}
+
+async fn route_and_handle(
+    router: Arc<Router>,
+    req_body: Request<hyper::Body>,
+    app_state: Arc<Mutex<AppState>>,
+) -> Result<Response, Error> {
+    let found_handler = router.route(req_body.uri().path(), req_body.method());
+    let resp = found_handler
+        .handler
+        .invoke(Context::new(app_state, req_body, found_handler.params))
+        .await;
+    Ok(resp)
+}
+
+async fn run_heartbeat_service(app: Arc<Mutex<AppState>>) {
+    const KILL_TIMEOUT: u64 = 5000;
+    println!("Heartbeat service running");
+
+    loop {
+        thread::sleep(time::Duration::from_millis(KILL_TIMEOUT));
+        let dead_users = app.lock().unwrap().get_dead_users();
+
+        app.lock().unwrap().clients.values_mut()
+            .for_each(|user| user.is_alive = false); // flip users' status to dead
+
+        if dead_users.is_empty() {
+            println!("Everybody is alive");
+        }
+        else {
+            println!("Some clients died");
+            for dead_user_id in dead_users {
+                app.lock().unwrap().disconnect_user_from_all(dead_user_id);
+                // TODO: jak się to usunie, a user potem znowu włączy apkę, to skutkuje to
+                // wysłaniem przez serwer goodbye-message'a ponownie. Ja bym np stworzył jakiegoś
+                // enuma np UserState i tam trzymał 3 stany: Aktywny, Niekatywny i Martwy. No bo
+                // ustawienie z powrotem `is_alive = true` tutaj byłoby jakieś mega głupie
+                app.lock().unwrap().remove_user(dead_user_id);
+            }
+        }
+    }
+}
+
+async fn run_ws(app: Arc<Mutex<AppState>>) {
+    println!("Preparing WS...");
+
+    let ws_route = warp::path("ws")
+        .and(warp::ws())
+        .and(warp::any().map(move || app.clone()))
+        .and_then(handler::registration_handler);
+
+    let routes = ws_route.with(warp::cors().allow_any_origin());
+    println!("WS open on {}", WS_ADDR);
+    warp::serve(routes).run(([127, 0, 0, 1], 8000)).await;
+}
+
+async fn run_http(app: Arc<Mutex<AppState>>) {
+    println!("Preparing HTTP...");
+
+    let new_service = make_service_fn(move |_| {
+        let app_capture = app.clone();
+        async {
+            Ok::<_, Error>(service_fn(move |req| {
+                let router = app_capture.deref().lock().unwrap().routing_map.clone();
+                route_and_handle(router, req, app_capture.clone())
+            }))
+        }
+    });
+
+    let addr = "127.0.0.1:8080"
```

Comment:
Jeden adres zahardkodowany na górze, a ten tutaj. A spodziewam się, że powinno być to to samo ip

Reply:
Innymi słowy, należy to przenieść do wspólnego miejsca, a także umożliwić podanie tego jako argument przez CLI